react/CreditCard.tsx

```diff
@@ -0,0 +1,282 @@
+      document.head.querySelectorAll<HTMLLinkElement>('link[rel=stylesheet]')
+    ).map(link => link.href)
+
+    await postRobot.send(iframeRef.current!.contentWindow, 'setup', {
+      stylesheetsUrls,
+      paymentSystems: creditCardPaymentSystems,
+    })
+    setIframeLoading(false)
+  }, [creditCardPaymentSystems])
+
+  const getEncryptedCard = useCallback(async (): Promise<EncryptedCard | null> => {
+    const { data } = await postRobot.send(
+      iframeRef.current!.contentWindow,
+      'encryptedCard'
+    )
+    return data
+  }, [])
+
+  useEffect(function createPaymenteSystemListener() {
+    const listener = postRobot.on(
+      'paymentSystem',
+      ({ data }: { data: PaymentSystem }) => {
+        setSelectedPaymentSystem(data)
+      }
+    )
+    return () => listener.cancel()
+  }, [])
+
+  const validateDoc = () => {
+    if (!doc.value) {
+      setDoc({
+        ...doc,
+        showError: true,
+        error: true,
+        errorMessage: intl.formatMessage({
+          id: 'checkout-payment.input.requiredField',
+        }),
+      })
+      return false
+    }
+    if (doc.error) {
+      setDoc({
+        ...doc,
+        showError: true,
+        errorMessage: intl.formatMessage({
+          id: 'checkout-payment.input.requiredField',
+        }),
+      })
+      return false
+    }
+
+    return true
+  }
+
+  const handleSubmit = async () => {
+    const encryptedCard = await getEncryptedCard()
+    const docIsValid = validateDoc()
+    if (!selectedPaymentSystem || !encryptedCard || !docIsValid) {
+      return
+    }
+
+    setSaveLoading(true)
+    const paymentData = getPaymentData(
+      encryptedCard,
+      selectedPaymentSystem,
+      doc.value
+    )
+    const result = await savePaymentData([paymentData])
+
+    if (!result.error) {
+      setSavedCard(result.value[0])
+    }
+    setSaveLoading(false)
+  }
+
+  const handleChangeDoc = (data: any) => {
+    setDoc({
+      value: data.document,
+      error: !data.isValid,
+      errorMessage: '',
+      showError: false,
+    })
+  }
+
+  if (isSSR) {
+    return null
+  }
+
+  if (savedCard) {
+    return <SavedCard lastDigits={savedCard.lastDigits} />
+  }
+
+  return (
+    <div className="relative w-100">
+      {iframeLoading && (
+        <div className="absolute top-0 left-0 right-0 bottom-0 bg-white-70 z-1 flex items-center justify-center">
+          <Spinner />
+        </div>
+      )}
+      <div style={{ width: '100%' }}>
+        <iframe
+          title="card-form-ui"
+          // Using min-width to set the width of the iFrame, works around an issue in iOS that can prevent the iFrame from sizing correctly.
+          style={{ minWidth: '100%', minHeight: '200px' }}
+          // The scrolling attribute is set to 'no' in the iFrame tag, as older versions of IE don't allow this to be turned off in code and can just slightly add a bit of extra space to the bottom of the content that it doesn't report when it returns the height.
```

Comment:
please, split this comment into more lines. It's hard to read it when its so long